File: Min. USB Devices Ever Mounted/usb.rb
#
#   Minimum number of USB devices ever mounted, Ruby
#   Module written by Brandon Arvanaghi
#   Website: arvanaghi.com 
#   Twitter: @arvanaghi
#

require 'win32/registry'

minimumUSBHistory = 2;

if !ARGV.empty?
  minimumUSBHistory = ARGV[0].to_i
end

usbHistory = Win32::Registry::HKEY_LOCAL_MACHINE.open('SYSTEM\ControlSet001\Enum\USBSTOR').keys.length

if usbHistory >= minimumUSBHistory
  puts "Proceed!"
else
  puts "Number of USB devices ever mounted: " + usbHistory.to_s
end

File: Min. USB Devices Ever Mounted/usb.pl
#
#   Minimum number of USB devices ever mounted, Perl
#   Module written by Brandon Arvanaghi 
#   Website: arvanaghi.com 
#   Twitter: @arvanaghi
#

use Win32::TieRegistry( Delimiter=>"\\");

my $MinimumUSBHistory = 2;
if (@ARGV >= 1) {
	$MinimumUSBHistory = $ARGV[0];
}

$BaseKey = $Registry->Open("HKEY_LOCAL_MACHINE\\SYSTEM\\ControlSet001\\Enum\\USBSTOR", { Access=>Win32::TieRegistry::KEY_READ()|0x100 });

if (scalar(keys %$BaseKey) >= $MinimumUSBHistory) {
	print "Proceed!\n";
} else {
	print "Number of USB devices ever mounted: " .scalar(keys %$BaseKey) . "\n";
}